qiskit/dagcircuit/dagcircuit.py

```diff
@@ -216,7 +216,17 @@ def add_calibration(self, gate, qubits, schedule, params=None):
             Exception: if the gate is of type string and params is None.
         """
         if isinstance(gate, Gate):
-            self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
+            try:
+                gate_has_param_exp = isinstance(gate.params[0], ParameterExpression)
+            except IndexError:
+                gate_has_param_exp = False
+
+            if gate_has_param_exp:
+                self._calibrations[gate.name][
+                    (tuple(qubits), tuple([float(gate.params[0]._symbol_expr)]))
+                ] = schedule
+            else:
+                self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
```

Comment:
This looks very targetted at only gates with a single parameter.  We need to be general if we're including something like this.  Also, there's no need to access private members of the class - `float(gate.params[0])` is sufficient.

Reply:
Simplified this code so it's not dependent on number of parameters.